vendor/plugins/edu.harvard.dce.paella.singleVideoToggle/singleVideoToggle.js

```diff
@@ -10,40 +11,76 @@ paella.addPlugin(function () {
   return class SingleVideoTogglePlugin extends paella.ButtonPlugin {
     constructor () {
       super ();
+      this._toggleSingleVideoIndex = 0;
+      this._toggleHlsLiveResV1Index = 1;
+      this._toolTipOpts = ["Switch videos", "Change resolution"];
+      this._subClassOpts = ["showViewModeButton", "showMultipleQualitiesPlugin"];
+      this._iconClassOpts = ["icon-presentation-mode","icon-screen"];
+      this._curOptIndex = 0; // iOS single video: toggle 2 videos to show only one
       this._iOSProfile = 'one_big';
       this._masterVideo = null;
       this._toggleIndex = 1; //toggle to presentation when button pressed first time
     }
     getDefaultToolTip () {
-      return base.dictionary.translate("Switch videos");
+      return base.dictionary.translate(this._toolTipOpts[this._getOptIndex()]);
     }
     getAriaLabel() {
-      return base.dictionary.translate("Switch videos");
+      return base.dictionary.translate(this._toolTipOpts[this._getOptIndex()]);
     }
     getAlignment() {
       return 'right';
     }
     getSubclass() {
-      return "showViewModeButton";
+      // #DCE OPC-497 show current res text on button (get the mutli quality button style)
+      return base.dictionary.translate(this._subClassOpts[this._getOptIndex()]);
     }
     getIconClass() {
-      return 'icon-presentation-mode';
+      // #DCE OPC-497 show current res text on button
+      return base.dictionary.translate(this._iconClassOpts[this._getOptIndex()]);
     }
     getIndex() {
       return 450;
     }
     getInstanceName() {
       return "singleVideoTogglePlugin";
     }
-    getName () {
+    getName() {
       return "edu.harvard.dce.paella.singleVideoTogglePlugin";
     }
-    _currentPlayerProfile () {
+
+    _getOptIndex() {
+        return (paella.dce && paella.dce.hlsLiveToggleV1) ? this._toggleHlsLiveResV1Index : this._toggleSingleVideoIndex;
+    }
+
+    _currentPlayerProfile() {
       return paella.player.selectedProfile;
     }
+
+    _printResIfHlsLive(streams) {
+      // Check if hlsLiveToggleV1 flag exists and the path to the source resolution value exists
+      // Then add the current resolution value into the control button
+      if  (paella.dce && paella.dce.hlsLiveToggleV1 && streams
+          && streams.length > 0
+          && streams[0].sources
+          && streams[0].sources.hls
+          && streams[0].sources.hls.length > 0
+          && streams[0].sources.hls[0].res) {
+            this.setText(streams[0].sources.hls[0].res.w + "p"); // change the UI res text
```

Comment:
This works if we use different resolution and not just change bandwidth, but we can cross that bridge if we even come to it.

Reply:
Right. Paella is expecting different resolutions. It doesn't check the bandwidth. So its more than this piece of code that would have to adapt to that in the future.